<?xml version="1.0"?>

<!-- joint FoR = from the orgin of the parent part
origin FoR = from the origin of the child part -->

<robot name="rhr_hand_basic">
  <link name="base_link">
    <visual>
      <geometry>
        <!-- <cylinder length="0.05" radius="0.05"/> -->
        <mesh filename="package://Palm/meshes/Palm.STL"/>
      </geometry>
      <!-- <origin rpy="0 0 0" xyz="0 0 0"/> -->
      <origin rpy="1.5708 0 1.5708" xyz="0 0 0"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
  </link>

  <link name="prox1">
    <visual>
      <geometry>
        <!-- <box size="0.06 .02 .02"/> -->
        <mesh filename="package://ProximalLink/meshes/ProximalLink.STL"/>
      </geometry>
      <!-- <origin rpy="0 0 0" xyz="0.03 0 0"/> -->
      <origin rpy="1.5708 0 1.5708" xyz="0.03 0 0"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>

  <joint name="base_to_prox1" type="revolute">
    <axis xyz="0 1 0"/>
    <parent link="base_link"/>
    <child link="prox1"/>
    <limit effort="1000" lower="-2.8" upper="0" velocity="1"/>
    <origin rpy="0 0 0" xyz="0.05 0 0.01"/>
  </joint>

  <link name="dist1">
    <visual>
      <geometry>
        <!-- <box size=".04 0.01 .01"/> -->
        <mesh filename="package://DistalLink/meshes/DistalLink.STL"/>
      </geometry>
      <origin rpy="1.5708 0 1.5708" xyz="0.02 0 0"/>
      <!-- <origin rpy="0 0 0" xyz="0.02 0 0"/> -->
      <material name="green">
        <color rgba="0 1 0 1"/>
      </material>
    </visual>
  </link>

  <joint name="prox1_to_dist1" type="revolute">
    <axis xyz="0 1 0"/>
    <parent link="prox1"/>
    <child link="dist1"/>
    <limit effort="1000" lower="-2.8" upper="0" velocity="1"/>
    <origin rpy="0 0 0" xyz="0.06 0 0"/>
  </joint>

  <link name="prox2">
    <visual>
      <geometry>
        <!-- <box size="0.06 .02 .02"/> -->
        <mesh filename="package://ProximalLink/meshes/ProximalLink.STL"/>
      </geometry>
      <origin rpy="1.5708 0 -1.5708" xyz="-0.03 0 0"/>
      <!-- <origin rpy="0 0 0" xyz="-0.03 0 0"/> -->
      <material name="white"/>
    </visual>
  </link>

  <joint name="base_to_prox2" type="revolute">
    <axis xyz="0 1 0"/>
    <parent link="base_link"/>
    <child link="prox2"/>
    <limit effort="1000" lower="0" upper="2.8" velocity="1"/>
    <origin rpy="0 0 0" xyz="-0.05 -0.025 0.01"/>
  </joint>

  <link name="dist2">
    <visual>
      <geometry>
        <!-- <box size=".04 0.01 .01"/> -->
        <mesh filename="package://DistalLink/meshes/DistalLink.STL"/>
      </geometry>
      <origin rpy="1.5708 0 -1.5708" xyz="-0.02 0 0"/>
      <!-- <origin rpy="0 0 0" xyz="-0.02 0 0"/> -->
      <material name="green"/>
    </visual>
  </link>

  <joint name="prox2_to_dist2" type="revolute">
    <axis xyz="0 1 0"/>
    <parent link="prox2"/>
    <child link="dist2"/>
    <limit effort="1000" lower="0" upper="2.8" velocity="1"/>
    <origin rpy="0 0 0" xyz="-0.06 0 0"/>
  </joint>

  <link name="prox3">
    <visual>
      <geometry>
        <!-- <box size="0.06 .02 .02"/> -->
        <mesh filename="package://ProximalLink/meshes/ProximalLink.STL"/>
      </geometry>
      <origin rpy="1.5708 0 -1.5708" xyz="-0.03 0 0"/>
      <!-- <origin rpy="0 0 0" xyz="-0.03 0 0"/> -->
      <material name="white"/>
    </visual>
  </link>

  <joint name="base_to_prox3" type="revolute">
    <axis xyz="0 1 0"/>
    <parent link="base_link"/>
    <child link="prox3"/>
    <limit effort="1000" lower="0" upper="2.8" velocity="1"/>
    <origin rpy="0 0 0" xyz="-0.05 0.025 0.01"/>
  </joint>

  <link name="dist3">
    <visual>
      <geometry>
        <!-- <box size=".04 0.01 .01"/> -->
        <mesh filename="package://DistalLink/meshes/DistalLink.STL"/>
      </geometry>
      <origin rpy="1.5708 0 -1.5708" xyz="-0.02 0 0"/>
      <!-- <origin rpy="0 0 0" xyz="-0.02 0 0"/> -->
      <material name="green"/>
    </visual>
  </link>

  <joint name="prox3_to_dist3" type="revolute">
    <axis xyz="0 1 0"/>
    <parent link="prox3"/>
    <child link="dist3"/>
    <limit effort="1000" lower="0" upper="2.8" velocity="1"/>
    <origin rpy="0 0 0" xyz="-0.06 0 0"/>
  </joint>

</robot>

